.github/workflows/stalebot.yml

```diff
@@ -16,25 +13,17 @@ jobs:
       issues: write
       pull-requests: write
     steps:
-      - name: Install Actions Core
-        run: npm --prefix .github/workflows/scripts install @actions/core
-
-      - name: Get start date
-        id: startdate
-        run: node .github/workflows/scripts/stalebot.js
-        env:
-          CRON_SCHEDULE: ${{ github.event.schedule }}
       - name: Scan issues
-        uses: actions/stale@v8
+        uses: actions/stale@b69b346013879cedbf50c69f572cd85439a41936
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-          operations-per-run: 8
-          start-date: steps.startdate.outputs.stale-start-date
+          operations-per-run: 1
```

Comment:
Is this supposed to be `100`?

Reply:
The iniitial api query retrieves a page of 100 items. `operations-per-run` might be better named as `batches-per-run` where each operation/batch is the 100 items retrieved in the query.

Reply:
The [log for the successful v8 run](https://github.com/woocommerce/woocommerce/actions/runs/6739800498/job/18321918187#step:2:561) has ~4 lines per issue processed and 100 operations complete at line 460 which suggests that it will process close to 100 issues per operation. Issues that are updated may require additional api calls but we shouldn't see more than 10 new stale items per day once this is up and running.

Reply:
> The iniitial api query retrieves a page of 100 items. operations-per-run might be better named as batches-per-run where each operation/batch is the 100 items retrieved in the query.

Looking through [this source file](https://github.com/actions/stale/blob/b69b346013879cedbf50c69f572cd85439a41936/src/classes/issues-processor.ts#L146) paints a clearer picture. Although it fetches 100 issues, if it runs out of operations while processing a batch, it will stop processing issues.

>  Issues that are updated may require additional api calls but we shouldn't see more than 10 new stale items per day once this is up and running.

As far as I can tell, these types of operations will take place:

* All Issues
  * One operation per batch to pull the 100 issues.
* Marking Stale
  * One operation to post the stale comment.
  * One operation to add the stale label.
  * N operations to remove and add any labels required as per a state transition.
* Stale Issues
  * One operation getting the date that the label was added.
  * One operation getting the comments made since the last.
  * **Oddly enough, the above two operations also happen to issues marked as stale.**
  * One operation when an issue is no longer stale
  * N operations to remove and add any labels required as per a state transition.
  * One operation to close the issue if it's time to do so
  * One operation to add a close comment if we're closing the issue.
  * One operation to add a label if one is set when closing.
  * One operation if the issue is no longer stale.

It should take 20 operations per cycle through all of our issues, and given how few issues will be marked as stale, probably only another 200 when it marks them as stale and 100 each time it checks them. Once they're closed, that'll be another 200 or so.

Based on all of this, 100 operations is probably a perfectly reasonable number. It doesn't do _any_ extra operations for non-stale issues since it gets information about the labels and update date without more API calls. There was a bug with PRs that made each of them consume an extra operation but that's fixed in the latest commit we're pinning. We will see very small spikes when marking and closing issues but that's fine.

Reply:
@ObliviousHarmony Thanks for documenting that.

Reply:
> it looks like we need to set operations-per-run to 100 in order for it to work correctly.

When `operations-per-run` was 40 it processed 847 issues/PRs before running into the api limit. So, we know that operations per run has to be more than a 20 => 1 factor. Based on the log it approaches 100 => 1.

Reply:
Keep in mind though that stale issues (and PRs in the future likely!) take up more operations than just paginating the list does. A limit of `20` scrolls you through the repository by itself, but if there are stale issues, each of them consumes an extra two operations. When you transition an issue or PR it also takes up a bunch of extra operations. Depending on the length of time we let things sit as stale too, this could quickly consume a lot of operations.

You're not wrong that a limit below `100` is probably fine, but a limit of `1` will read 100 issues and then break without actually processing any of them. You can see [here in the source](https://github.com/actions/stale/blob/b69b346013879cedbf50c69f572cd85439a41936/src/classes/issues-processor.ts#L146) that it stops processing issues when no operations remain.

Reply:
How about I remove the parameter and we can look at the log for the default of 30?

Reply:
Deal

Reply:
Removed `operations-per-run`.